src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -118,6 +116,23 @@ public virtual bool CanValidateToken
             get { return true; }
         }
 
+        private int CountTokenSegments(string token)
```

Comment:
The name of the method is technically misleading since it breaks when number of '.''s is greater than JwtConstants.MaxJwtSegmentCount +1.